File: src/main/java/com/github/xumeng/java/problems/easy/problem0283/Solution0283S2.java
package com.github.xumeng.java.problems.easy.problem0283;

public class Solution0283S2 {

    public void moveZeroes(int[] nums) {
        int length = nums.length;
        for (int i = 0; i < length; i++) {
            for (int j = 0; j + 1 < length - i; j++) {
                if (nums[j] != 0) {
                    continue;
                }
                if (nums[j + 1] == 0) {
                    continue;
                }
                int temp = nums[j + 1];
                nums[j + 1] = nums[j];
                nums[j] = temp;
            }
        }
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0283/Solution0283S2.kt
package com.github.xumeng.kotlin.problems.easy.problem0283

class Solution0283S2 {

    fun moveZeroes(nums: IntArray) {
        val length: Int = nums.size
        var temp: Int
        for (i in 0..<length) {
            for (j in 0..<length - 1 - i) {
                if (nums[j] != 0) {
                    continue
                }
                if (nums[j + 1] == 0) {
                    continue
                }
                temp = nums[j + 1]
                nums[j + 1] = nums[j]
                nums[j] = temp
            }
        }
    }

}